evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractCallOperation.java

```diff
@@ -349,19 +379,22 @@ Bytes getCallResultStackItem(final MessageFrame childFrame) {
    * Gets the code from the contract or EOA with delegated code.
    *
    * @param evm the evm
-   * @param account the account which needs to be retrieved
+   * @param frame the message frame
+   * @param account the account which codes needs to be retrieved
    * @return the code
    */
-  protected static Code getCode(final EVM evm, final Account account) {
+  protected static Code getCode(final EVM evm, final MessageFrame frame, final Account account) {
     if (account == null) {
       return CodeV0.EMPTY_CODE;
     }
 
-    if (account.hasDelegatedCode()) {
-      return evm.getCode(
-          account.getCodeDelegationTargetHash().get(), account.getCodeDelegationTargetCode().get());
+    if (!hasCodeDelegation(account.getCode())) {
```

Comment:
why we are not also checking empty as here ? https://github.com/hyperledger/besu/pull/8408/files#diff-80136cfd3cd03f87ae7a2dee65ff1a49be6a02921656313b6d70437fc932eb52R637

Reply:
I've added the check